lib/aker_permission_client_config.rb

```diff
@@ -0,0 +1,33 @@
+module AkerPermissionClientConfig
+  def has_permission?(user_email, role)
+    mapping = {read: :r, write: :w, execute: :x}    
+    permissions.any? do |permission|
+      (permission.permitted == user_email) && (permission.send(mapping[role]))
+    end
+  end
+
+  def self.included(base)
+    base.instance_eval do
+    def self.authorize!(role, resource, user_email)
+      raise CanCan::AccessDenied.new("Not authorized!", role, resource) unless user_email
```

Comment:
If user_email is falsey, I think this would be an ArgumentError, rather than AccessDenied.